core/main/src/service/apps/pending_session_event_processor.rs

```diff
@@ -0,0 +1,81 @@
+use ripple_sdk::extn::client::extn_processor::ExtnStreamer;
+use ripple_sdk::{
+    api::app_catalog::AppsUpdate,
+    async_trait::async_trait,
+    extn::{
+        client::extn_processor::{DefaultExtnStreamer, ExtnEventProcessor, ExtnStreamProcessor},
+        extn_client_message::ExtnMessage,
+    },
+    log::error,
+    tokio::sync::mpsc::Sender,
+};
+
+use crate::{
+    service::apps::delegated_launcher_handler::DelegatedLauncherHandler,
+    state::{cap::permitted_state::PermissionHandler, platform_state::PlatformState},
+};
+
+pub struct PendingSessionEventProcessor {
+    platform_state: PlatformState,
+    streamer: DefaultExtnStreamer,
+}
+
+impl PendingSessionEventProcessor {
+    pub fn new(platform_state: PlatformState) -> PendingSessionEventProcessor {
+        PendingSessionEventProcessor {
+            platform_state,
+            streamer: DefaultExtnStreamer::new(),
+        }
+    }
+}
+
+impl ExtnStreamProcessor for PendingSessionEventProcessor {
+    type STATE = PlatformState;
+    type VALUE = AppsUpdate;
+    fn get_state(&self) -> Self::STATE {
+        self.platform_state.clone()
+    }
+
+    fn sender(&self) -> Sender<ExtnMessage> {
+        self.streamer.sender()
+    }
+
+    fn receiver(&mut self) -> ripple_sdk::tokio::sync::mpsc::Receiver<ExtnMessage> {
+        self.streamer.receiver()
+    }
+}
+
+#[async_trait]
+impl ExtnEventProcessor for PendingSessionEventProcessor {
+    async fn process_event(
+        state: Self::STATE,
+        _msg: ExtnMessage,
+        extracted_message: Self::VALUE,
+    ) -> Option<bool> {
+        if let AppsUpdate::InstallComplete(operation) = extracted_message.clone() {
+            let has_pending_session = state.session_state.has_pending_session(&operation.id);
+
+            if has_pending_session {
+                if operation.success {
+                    // Send request to create session and send lifecyclemanagement.OnSessionTransitionComplete
+                    match PermissionHandler::fetch_permission_for_app_session(&state, &operation.id)
+                        .await
+                    {
+                        Ok(()) => {
+                            DelegatedLauncherHandler::emit_completed(&state, &operation.id).await
```

Comment:
doesn't there need to be more work for delegated launcher handler to do after it fetches the permissions? For example, it needs to check the grant states again. Basically everything that happens after the return statement you added needs to start executing again